### PR TITLE
7272: Removal of 1995s from Spool files and YTD reports

### PR DESCRIPTION
**Pull request by elder-joseph**
## Description of change
As a veteran looking to update my benefits, I should not have to answer unnecessary questions about the STEM scholarship application in order to update my benefits. 

This Pull Request is associated with the `vets-website` pull request [12120](https://app.zenhub.com/workspaces/vft-59c95ae5fda7577a9b3184f8/issues/department-of-veterans-affairs/vets-website/12120)

[Originating Issue](https://app.zenhub.com/workspaces/vft-59c95ae5fda7577a9b3184f8/issues/department-of-veterans-affairs/va.gov-team/7272)

## Testing
Unit tests pass locally after making the updates

## Acceptance Criteria (Definition of Done)
- [x] The 22-1995 spool file does not include STEM Scholarship Application submissions.
- [x] The YTD Report does not include 22-1995 STEM information

#### Unique to this PR
This PR is implemented behind the `edu_benefits_stem_scholarship` for toggling the STEM scholarship functionality

#### Applies to all PRs

- [x] Swagger docs have been updated, if applicable
- [x] Provide link to originating GitHub issue, or connected to it via ZenHub
- [x] Does not contain any sensitive information (i.e. PII/credentials/internal URLs/etc., in logging, hardcoded, or in specs)
- [x] Provide which alerts would indicate a problem with this functionality (if applicable)

### CODE REVIEW

File: db/schema.rb [hunk 1]
```diff
@@ -139,6 +139,25 @@
...
+  end
+
+  create_table "disability_compensation_submissions", id: :serial, force: :cascade do |t|
+    t.datetime "created_at", null: false
+    t.datetime "updated_at", null: false
+    t.integer "disability_compensation_id"
+    t.integer "va526ez_submit_transaction_id"
+    t.boolean "complete", default: false
+  end
+
```
Comment (jholton) [inline db/schema.rb:160]: Revert these changes please.  This was unintentional, right?
I've noticed this `schema.rb` change happens when running `vets-api` via Docker.
I'm going to try to figure out why.

Comment (elder-joseph) [reply db/schema.rb:160]: Hey @jholton I will revert that, but yes it does seem to be to an unintended consequence of running `vets-api` via Docker

File: app/workers/education_form/templates/1995.erb [hunk 2]
```diff
@@ -29,13 +29,6 @@ Preferred Method of Contact: <%= @applicant.preferredContactMethod %>
                   TYPE AND PROGRAM OF EDUCATION OR TRAINING
                   -----------------------------------------
 
-Benefit You Are Receiving: <%= form_benefit %>
```
Comment (jholton) [inline app/workers/education_form/templates/1995.erb:32]: Do you want this line gone?  It's not STEM specific and [was present before the STEM stuff was added](https://github.com/department-of-veterans-affairs/vets-api/blame/71b87a85784cc236710717dc823d8b6ab2266a28/app/workers/education_form/templates/1995.erb).

File: spec/jobs/education_form/create_daily_fiscal_year_to_date_report_spec.rb [hunk 3]
```diff
@@ -72,9 +72,16 @@ def get_education_form_fixture(filename)
       create(:education_benefits_submission, created_at: date - 26.hours, status: 'processed')
 
       create(:education_benefits_submission, created_at: date, status: 'submitted')
-      %w[1995 1990e 5490 1990n 5495 1995s].each do |form_type|
-        create(:education_benefits_submission, form_type: form_type, created_at: date)
+      if !Flipper.enabled?(:edu_benefits_stem_scholarship, @current_user)
```
Comment (jholton) [inline spec/jobs/education_form/create_daily_fiscal_year_to_date_report_spec.rb:75]: We shouldn't be checking if `Flipper.enabled?` in the specs.
At this point when the specs run is `Flipper.enabled?(:edu_benefits_stem_scholarship, @current_user)` true or false?
Regardless, there is one path that does not get tested here.

Instead you should set the toggle value to what you want it to be and test as appropriate.
You can search the `spec` directory for `Flipper.enable` to find examples of other devs doing this.  [Here's one](https://github.com/department-of-veterans-affairs/vets-api/blob/master/spec/lib/facilities/ppms/client_spec.rb).

**Comment by jholton** [comment]
After this is deployed and the feature toggle is disabled, is form `1995s` completely going away?
If so, do you have issues written to remove its related files?

File: app/models/education_benefits_claim.rb [hunk 4]
```diff
@@ -2,10 +2,17 @@
...
+    FORM_TYPES = %w[1990 1995 1990e 5490 5495 1990n 0993 0994].freeze
+    FORM_HEADERS = %w[
+      22-1990 22-1995 22-1990e 22-5490 22-5495 22-1990n 22-0993 22-0994
+    ].freeze
+  else
+    FORM_TYPES = %w[1990 1995 1990e 5490 5495 1990n 0993 0994 1995s].freeze
+    FORM_HEADERS = [
+      '22-1990', '22-1995', '22-1990e', '22-5490', '22-5495', '22-1990n', '22-0993', '22-0994', '22-1995 STEM'
+    ].freeze
+  end
```
[the remark this answers is not in the document]
Comment (elder-joseph) [reply app/models/education_benefits_claim.rb:15]: I see that, no problem it is a little more straight forward without the inversion.

File: app/workers/education_form/templates/1995.erb (hunk 2 above)
Comment (elder-joseph) [reply app/workers/education_form/templates/1995.erb:32]: I think that we do want to add that back in, I will do so.

File: spec/jobs/education_form/create_daily_fiscal_year_to_date_report_spec.rb (hunk 3 above)
Comment (elder-joseph) [reply spec/jobs/education_form/create_daily_fiscal_year_to_date_report_spec.rb:75]: Thank you for sending over the example of that. But yes that is clear I will add in specs for the other case here.

File: app/models/education_benefits_claim.rb (hunk 4 above)
[the remark this answers is not in the document]
Comment (zurbergram) [reply app/models/education_benefits_claim.rb:15]: Inversion is not required as this logic is incorrect, simply dropping the `!` will use flag correctly

**Comment by zurbergram** [comment]
The logic for the feature flag is:

if `edu_benefits_stem_scholarship`

- vets-website
  - 10203 is available to access through wizard
  - 1995 should not show 1995 STEM questions/eligibility
- vets-api
  - should not have 1995s or 22-1995 STEM as possible form type or header

if not edu_benefits_stem_scholarship
- vets-website
  - 10203 is not available to access through wizard
  - 1995 continues to show 1995 STEM questions/eligibility
- vets-api
  - should have 1995s or 22-1995 STEM as possible form type or header

**Comment by elder-joseph** [comment]
> After this is deployed and the feature toggle is disabled, is form `1995s` completely going away?
> If so, do you have issues written to remove its related files?

Hey @jholton The 1995s is being phased out of the regular 1995 workflow to be replaced with 10203.  EDU has requested that when the 10203 goes live the 1995s is disabled simultaneously.  We will have a follow-on story once everything is live and stable to remove the legacy 1995s code. Additionally, we have a follow on story to remove the associated feature flag.

File: spec/jobs/education_form/create_spool_submissions_report_spec.rb [hunk 5]
```diff
@@ -40,12 +40,12 @@
...
 
-      it 'recognizes 1995s as STEM submission' do
-        create(:education_benefits_claim_1995s, processed_at: time.beginning_of_day)
-        data = subject.create_csv_array
-        expect(data[:stem_exists]).to eq(true)
-      end
+    it 'recognizes 1995s as STEM submission' do
+      create(:education_benefits_claim_1995s, processed_at: time.beginning_of_day)
+      data = subject.create_csv_array
+      expect(data[:stem_exists]).to eq(true)
```
Comment (jholton) [inline spec/jobs/education_form/create_spool_submissions_report_spec.rb:48]: Should this be reverted?  Looks like all that has changed now is the moving of an `end`, which moves `it 'recognizes 1995s as STEM submission'` outside of the `describe '#create_csv_array'` block

Comment (elder-joseph) [reply spec/jobs/education_form/create_spool_submissions_report_spec.rb:48]: Hi @jholton that should be reverted, let me go ahead and do so.

File: app/models/education_benefits_claim.rb [hunk 6]
```diff
@@ -2,10 +2,17 @@
 
 require 'attr_encrypted'
 class EducationBenefitsClaim < ApplicationRecord
-  FORM_TYPES = %w[1990 1995 1990e 5490 5495 1990n 0993 0994 1995s].freeze
-  FORM_HEADERS = [
-    '22-1990', '22-1995', '22-1990e', '22-5490', '22-5495', '22-1990n', '22-0993', '22-0994', '22-1995 STEM'
-  ].freeze
+  if Flipper.enabled?(:edu_benefits_stem_scholarship, @current_user)
```
Comment (jholton) [inline app/models/education_benefits_claim.rb:5]: Are you enabling this for everyone, or just a percentage of users?  I don't think you need the @current_user parameter if you are enabling for all.
```suggestion
  if Flipper.enabled?(:edu_benefits_stem_scholarship)
```

**Comment by elder-joseph** [comment]
@jholton I committed your suggestion to enable the flag for all users. I cannot think of a use case where we would only want this feature flag enabled for a group of users.
